services/graphql-server/extensions/endpoints/endpoint.go

```diff
@@ -0,0 +1,63 @@
+package endpoints
+
+import (
+	"net/http"
+
+	authproviders "agogos/extensions/authproviders"
+	gqlconfig "agogos/generated"
+)
+
+// Endpoint - interface for endpoint extension. Allows to change @http directive url or to add headers to http request created in resolver
+type Endpoint interface {
+	ApplyEndpoint(req *http.Request)
+}
+
+type endpointStruct struct {
+	host    string
+	headers map[string]string
+	auth    authStruct
+}
+
+type authStruct struct {
+	authType  string
+	authority string
+	scope     string
+}
+
+func (ep *endpointStruct) ApplyEndpoint(req *http.Request) {
+	for header, headerValue := range ep.headers {
+		req.Header.Set(header, headerValue)
+	}
+
+	// TODO: consider more implicit approach
+	ap, ok := authproviders.Get(ep.auth.authority)
```

Comment:
Think how to decouple the `endpoint` extension from `authprovider` extension

Reply:
For this PR I'll leave the code as is (TODO comment will be taken care in future PRs)